### PR TITLE
fix: reused DAO sockets don't set the keyspace. fix #170

### LINKED ISSUE #170: Cassandra "set_keyspace" fails
Under load, Kong returns a bunch occurrences of the following exception:

```
nginx: 2015/04/28 09:20:22 [error] 24637#0: *492903 lua entry thread aborted: runtime error: /usr/local/share/lua/5.1/cassandra.lua:243: Result expected
stack traceback:
coroutine 0:
	[C]: in function 'error'
	/usr/local/share/lua/5.1/cassandra.lua:243: in function 'set_keyspace'
```
Does it need to try to set the keyspace every time? Can the operation be optimized?

### PR DESCRIPTION
**Pull request by thibaultCha**
DAO sockets now take advantage of the connection pool and if they are
coming from it (meaning they've been used before), they don't set their
keyspace anymore.

### CODE REVIEW

File: kong/dao/cassandra/base_dao.lua [hunk 1]
```diff
@@ -152,11 +152,18 @@ function BaseDao:_open_session()
...
   end
 
-  ok, err = session:set_keyspace(self._properties.keyspace)
-  if not ok then
+  local times, err = session:get_reused_times()
+  if err and err ~= "luasocket does not support reusable sockets" then
     return nil, DaoError(err, error_types.DATABASE)
   end
 
+  if times == 0 or not times then
```
Comment (lookfirst) [inline kong/dao/cassandra/base_dao.lua]: Not sure how lua works, but wouldn't you want to check for `not times` first?

Comment (thibaultCha) [inline kong/dao/cassandra/base_dao.lua]: Oh yeah that's a legacy check, only the 0 is actually needed, thanks.

Comment (thibaultCha) [inline kong/dao/cassandra/base_dao.lua]: Oops my bad, it was a still needed check. About the order, it is correct. Most of the time while in use, and in need of performance (when Kong will be running), `get_reused_times()` will return a number (0 possibly). And thus our first condition need to be that.

If we are in a phase of NGINX that doesn't support the cosocket API (`init_by_lua`) or maybe not even in the NGINX context (the DAO is also used by the CLI or by the unit tests), `get_reused_times()` can also error and return nil. That is a less common case and not in need of performance.

Hence why the check. My bad for thinking I could remove it, because I reorganized the code before pushing but overlooked at it. I closed this PR by mistake, and had to reopen it in #197 tho
